guides/source/action_view_overview.md

```diff
@@ -219,183 +279,166 @@ One way to use partials is to treat them as the equivalent of subroutines; a way
-  <h1><%= product.name %></h1>
+<%= tag.div id: dom_id(my_product) do %>
+  <h1><%= my_product.name %></h1>
 <% end %>
 ```
 
-If a template refers to a variable that isn't passed into the view as part of
-the `locals:` option, the template will raise an `ActionView::Template::Error`:
+A "partial-local variable" is a variable that is local to a given partial and
+only available from within that partial. In the above example, `my_product` is a
+partial-local variable. It was assigned the value of `@product` when passed to
+the partial from the original view.
+
+Note that typically we'd simply call this local variable `product`. We are using
+`my_product` to distinguish it from the instance variable name and template name
+in this example.
+
+Since `locals` is a hash, you can pass in multiple variables as needed, like
+`locals: { my_product: @product, my_reviews: @reviews }`.
+
+However, if a template refers to a variable that *isn't* passed into the view as
+part of the `locals:` option, the template will raise an
+`ActionView::Template::Error`:
 
 ```html+erb
 <%# app/views/products/_product.html.erb %>
 
-<%= tag.div id: dom_id(product) do %>
-  <h1><%= product.name %></h1>
+<%= tag.div id: dom_id(my_product) do %>
+  <h1><%= my_product.name %></h1>
 
-  <%# => raises ActionView::Template::Error %>
-  <% related_products.each do |related_product| %>
+  <%# => raises ActionView::Template::Error for `product_reviews` %>
+  <% product_reviews.each do |review| %>
     <%# ... %>
   <% end %>
 <% end %>
 ```
 
 ### Using `local_assigns`
 
-Each key in the `locals:` option is available as a partial-local variable through the [local_assigns][] helper method:
+Each partial has a method called [local_assigns][] available. You can use this
+method to access keys passed via the `locals:` option. If a partial was not
+rendered with `:some_key` set, the value of `local_assigns[:some_key]` will be
+`nil` within the partial.
+
+For example, `product_reviews` is `nil` in the below example since only
+`product` is set in `locals:`:
 
 ```html+erb
 <%# app/views/products/show.html.erb %>
 
-<%= render partial: "products/product", locals: { product: @product } %>
+<%= render partial: "product", locals: { product: @product } %>
 
 <%# app/views/products/_product.html.erb %>
 
 <% local_assigns[:product]          # => "#<Product:0x0000000109ec5d10>" %>
-<% local_assigns[:options]          # => nil %>
+<% local_assigns[:product_reviews]  # => nil %>
 ```
 
-Since `local_assigns` is a `Hash`, it's compatible with [Ruby 3.1's pattern matching assignment operator](https://docs.ruby-lang.org/en/master/syntax/pattern_matching_rdoc.html):
-
-```ruby
-local_assigns => { product:, **options }
-product # => "#<Product:0x0000000109ec5d10>"
-options # => {}
-```
-
-When keys other than `:product` are assigned into a partial-local `Hash`
-variable, they can be splatted into helper method calls:
+One use case for `local_assigns` is optionally passing in a local variable and
```

Comment:
Should we add a note that now this is also possibly using default values on strict locals?

Reply:
That's enough. TY!